### PR TITLE
network: routing policy rule cleanups

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by keszybz** [comment]
CI no like:
```
276/809 test-routing-policy-rule                FAIL     0.17 s (killed by signal 6 SIGABRT)

--- command ---
SYSTEMD_LANGUAGE_FALLBACK_MAP='/tmp/autopkgtest.1plWPg/build.E7G/systemd/src/locale/language-fallback-map' PATH='/tmp/autopkgtest.1plWPg/build.E7G/systemd/build-deb:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin' SYSTEMD_KBD_MODEL_MAP='/tmp/autopkgtest.1plWPg/build.E7G/systemd/src/locale/kbd-model-map' /tmp/autopkgtest.1plWPg/build.E7G/systemd/build-deb/test-routing-policy-rule
--- stdout ---
--- /tmp/systemd-test-routing-policy-rule.6vDyct	2020-08-28 05:30:46.060289127 +0000
+++ /tmp/systemd-test-routing-policy-rule.kny7Qm	2020-08-28 05:30:46.060289127 +0000
@@ -1 +1 @@
-RULE=from=1.2.3.4/32 to=2.3.4.5/32 family=AF_INET tos=5 priority=0 fwmark=1/2 table=10
+RULE=family=AF_INETfrom=1.2.3.4/32 to=2.3.4.5/32 tos=5 fwmark=1/2 invert_rule=no table=10
--- stderr ---
Failed to read $container of PID 1, ignoring: Permission denied
Found container virtualization none.
========== basic parsing ==========
put:
RULE=from=1.2.3.4/32 to=2.3.4.5/32 family=AF_INET tos=5 priority=0 fwmark=1/2 table=10
Unknown RPDB rule, ignoring: family
got:
RULE=family=AF_INETfrom=1.2.3.4/32 to=2.3.4.5/32 tos=5 fwmark=1/2 invert_rule=no table=10
$ diff -u /tmp/systemd-test-routing-policy-rule.6vDyct /tmp/systemd-test-routing-policy-rule.kny7Qm
Assertion 'system(cmd) == 0' failed at src/network/test-routing-policy-rule.c:54, function test_rule_serialization(). Aborting.
-------
```

**Comment by yuwata** [comment]
@keszybz Test is updated.

**Comment by keszybz** [comment]
LGTM.

File: src/network/networkd-routing-policy-rule.c [hunk 1]
```diff
@@ -1453,6 +1454,13 @@ int routing_policy_load_rules(const char *state_file, Set **rules) {
                                         log_error_errno(r, "Failed to parse RPDB rule suppress_prefixlen, ignoring: %s", b);
                                         continue;
                                 }
+                        } else if (streq(a, "invert_rule")) {
+                                r = parse_boolean(b);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to parse RPDB rule invert_rule, ignoring: %s", b);
```
Comment (poettering) [inline src/network/networkd-routing-policy-rule.c]: should be downgraded to warning i guess (and the other cases like this too). 

(i.e. if a log message says "ignoring" it's usually a good indication it should be warning, and not error)

File: src/network/networkd-routing-policy-rule.c [hunk 2]
```diff
@@ -1356,7 +1356,18 @@ int routing_policy_load_rules(const char *state_file, Set **rules) {
                         }
                         *b++ = '\0';
 
-                        if (STR_IN_SET(a, "from", "to")) {
+                        if (streq(a, "family")) {
+                                r = af_from_name(b);
+                                if (r < 0) {
```
Comment (poettering) [inline src/network/networkd-routing-policy-rule.c:1361]: as above

File: src/network/networkd-routing-policy-rule.c [hunk 3]
```diff
@@ -1356,7 +1356,18 @@ int routing_policy_load_rules(const char *state_file, Set **rules) {
...
+                                        log_error_errno(r, "Failed to parse RPDB rule family, ignoring: %s", b);
+                                        continue;
+                                }
+                                if (rule->family != AF_UNSPEC && rule->family != r) {
+                                        log_error("RPDB rule family is already specified, ignoring assignment: %s", b);
```
Comment (poettering) [inline src/network/networkd-routing-policy-rule.c]: here too, and everywhere else

**Review by poettering — APPROVED** [review]
minor nitpick about the log levels, but i guess this is pre-existing

**Comment by yuwata** [comment]
@poettering Thank you for the review, and sorry for late response. Now log level is downgraded. Setting the green label.

**Comment by poettering** [comment]
lgtm, thanks!

**Comment by keszybz** [comment]
Bad semaphore: `collect2: fatal error: ld terminated with signal 11 [Segmentation fault]`

**Comment by keszybz** [comment]
bionic-i386 timed out.

**Pull request merged**